modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -104,95 +105,220 @@ private void flattenPaths(OpenAPI openAPI) {
+            if (m.getAnyOf() != null && !m.getAnyOf().isEmpty()) {
+                return true;
+            }
+            if (m.getOneOf() != null && !m.getOneOf().isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
 
-        Schema model = ModelUtils.getSchemaFromRequestBody(requestBody);
-        if (model instanceof ObjectSchema) {
-            Schema obj = (Schema) model;
-            if (obj.getType() == null || "object".equals(obj.getType())) {
-                if (obj.getProperties() != null && obj.getProperties().size() > 0) {
-                    flattenProperties(openAPI, obj.getProperties(), pathname);
-                    // for model name, use "title" if defined, otherwise default to 'inline_object'
-                    String modelName = resolveModelName(obj.getTitle(), "inline_object");
-                    addGenerated(modelName, model);
-                    openAPI.getComponents().addSchemas(modelName, model);
-
-                    // create request body
-                    RequestBody rb = new RequestBody();
-                    rb.setRequired(requestBody.getRequired());
-                    Content content = new Content();
-                    MediaType mt = new MediaType();
-                    Schema schema = new Schema();
-                    schema.set$ref(modelName);
-                    mt.setSchema(schema);
-
-                    // get "consumes", e.g. application/xml, application/json
-                    Set<String> consumes;
-                    if (requestBody == null || requestBody.getContent() == null || requestBody.getContent().isEmpty()) {
-                        consumes = new HashSet<>();
-                        consumes.add("application/json"); // default to application/json
-                        LOGGER.info("Default to application/json for inline body schema");
-                    } else {
-                        consumes = requestBody.getContent().keySet();
+    /**
+     * Recursively gather inline models that need to be generated and
+     * replace inline schemas with $ref to schema to-be-generated.
+     *
+     * @param schema target schema
+     */
+    private void gatherInlineModels(Schema schema, String modelPrefix) {
+        if (schema.get$ref() != null) {
+            // if ref already, no inline schemas should be present but check for
+            // any to catch OpenAPI violations
+            if (isModelNeeded(schema) || "object".equals(schema.getType()) ||
+                    schema.getProperties() != null || schema.getAdditionalProperties() != null ||
+                    schema instanceof ComposedSchema) {
+                LOGGER.error("Illegal schema found with $ref combined with other properties," +
+                        " no properties should be defined alongside a $ref:\n " + schema.toString());
+            }
+            return;
+        }
+        if (schema.getType() == null || "object".equals(schema.getType())) {
```

Comment:
Why are we handling null type in here?
Isn't null type an AnyType schema?
Or is this detecting composed schemas with properties?
How about adding comments describing these use cases here?

Reply:
Good point, added this comment let me know if that's not clear:
```java
        // Check object models / any type models / composed models for properties,
        // if the schema has a type defined that is not "object" it should not define
        // any properties
```